### ISSUE TITLE
TypeError from htmlspecialchars when using array variable

### ISSUE THREAD

**Issue by nie7321** [bobthecow/mustache.php#400]
Running the following code on PHP 8.1.3:

```php
$engine = new Mustache_Engine;
$engine->render('{{ items }}', json_decode('{"items": ["aibi", "owls"]}'))
```

Results in `TypeError: htmlspecialchars(): Argument #1 ($string) must be of type string, array given`.

According to the spec & demo tool, this example should work and yield the result `aibi,owls`. From the site:

![image](https://user-images.githubusercontent.com/27235866/177383361-53b5537b-e9bc-4163-a60c-6137f2dc498b.png)

This was a warning in PHP 7.x, but in 8.0 it was upgraded to the TypeError exception: https://3v4l.org/IX2Bo

**Comment by bobthecow** [comment]
Behavior when interpolating an array as a string is undefined in the spec, and in such matters the spec defers to what is idiomatic in the implementing language.

The demo tool on the website is a javascript implementation, so what you're seeing is the idiomatic javascript behavior when casting an array to a string:

<img width="762" alt="Screen Shot 2022-07-06 at 8 36 12 AM" src="https://user-images.githubusercontent.com/53660/177551457-efcfaa9b-746e-4656-93ac-968aee6ebf8c.png">

Implicit coercion into a comma separated list has never been the idiomatic behavior in PHP, though: 

<img width="906" alt="Screen Shot 2022-07-06 at 8 40 26 AM" src="https://user-images.githubusercontent.com/53660/177552279-85cbd0be-bb53-4fe5-96f2-3a47aeec5440.png">

<img width="906" alt="Screen Shot 2022-07-06 at 8 42 49 AM" src="https://user-images.githubusercontent.com/53660/177552681-f2a0b021-bc7b-4155-a80c-2f99f0f320ff.png">

**Comment by nie7321** [comment]
Gotcha, thanks! 

I was looking at passing the escape function to handle it, but I think that's a bit nicer since it's explicit.

**Comment by bobthecow** [comment]
Yep! It's always better to tell the machine how to resolve ambiguity :)
